### ISSUE TITLE
IValueInsertable not using parameters ?!

### ISSUE THREAD

**Issue by drekoo** [linq2db/linq2db#1624]
Hi,

```
var insertStatus = conn.GetTable<POGODBA_STATUS>()
                .Value(v => v.POGODBA_ID, vsebina.POGODBA_ID)
                .Value(v => v.POGODBA_VSEBINA_ID, vsebinaID)
                .Value(v => v.DATUM, DateTime.Now)
                .Value(v => v.OPIS, status.OPIS)
                .Value(v => v.STATUS, status.STATUS);
 insertStatus.Insert();
```

**PRODUCES SQL WITHOUT PARAMETERS:**

```
INSERT INTO STATUS
(
	POGODBA_ID,
	POGODBA_VSEBINA_ID,
	DATUM,
	OPIS,
	STATUS
)
VALUES
(
	3938,
	9349,
	'2019-02-25 16:04:22.424',
	'Description',
	1
)
```

### Environment details
linq2db version: *2.6.4*
Database Server: *FIREBIRD*

**Comment by sdanyliv** [comment]
@drekoo, rewrite your code to use expressions. In this case engine will treat values as parameters.
```cs
var insertStatus = conn.GetTable<POGODBA_STATUS>()
                .Value(v => v.POGODBA_ID, () => vsebina.POGODBA_ID)
                .Value(v => v.POGODBA_VSEBINA_ID, () => vsebinaID)
                .Value(v => v.DATUM, () => DateTime.Now)
                .Value(v => v.OPIS, () => status.OPIS)
                .Value(v => v.STATUS, () => status.STATUS);
 insertStatus.Insert();
```

**Comment by drekoo** [comment]
I think, I have already tried that  with the same result output.... Will try again tomorrow at work.....

**Comment by drekoo** [comment]
I have checked. Result is same. NO PARAMETERS!

**Comment by sdanyliv** [comment]
Strange, because I have checked this before creating post. Check maybe somewhere in DataConnection initialization you have set `InlineParameters = true`

**Comment by drekoo** [comment]
I don't have  InlineParameters anywhere.

**Debug IValueInsertable variable sql seems ok,** 
![screen1](https://user-images.githubusercontent.com/38095833/53401624-787d0f80-39b0-11e9-87f4-8b8d532487fd.jpg)
**Actual insert execution is wrong**
![screen2](https://user-images.githubusercontent.com/38095833/53401644-82067780-39b0-11e9-8c22-5a3a11d13ead.jpg)

**Comment by sdanyliv** [comment]
Will check, maybe something wrong with provider.

**Comment by sdanyliv** [comment]
I confirm problem. We have to discuss why we have disabled parameters for Firebird. Maybe some limitations in previous versions.

**Comment by drekoo** [comment]
Ok thanks!
Just for info, when using "Entity/Object" insert, parameters are used. 
This is current workaround for blob insert, which is not working with IValueInsertable (consequently sql X"blob" size problem).

```
var entity = new Database.Entity.POGODBA_VSEBINA
                    {
                        NAZIV = fileName,
                        OPIS = fileNote,
                        POGODBA_ID = pogodbaId,
                        TIP = 2,
                        VSEBINA = memoryStream.ToArray()
                    };

conn.InsertWithInt32Identity(entity);
```

**Comment by MaceWindu** [comment]
I've looked into it - we currently convert parameters to literals for Firebird, because frankly speaking - Firebird's support for parameters is far from ideal and a lot of queries just fail with `-804 Data type unknown` when parameters used (or even with worse kinds of errors).

Don't think we can fix it, as it server limitation in most of cases.

**Comment by MaceWindu** [comment]
Fixed by #2152
